src/nwm_routing/src/nwm_routing/preprocess.py

```diff
@@ -206,36 +199,47 @@ def nwm_initial_warmstate_preprocess(
                 len(waterbodies_initial_states_df)
             )
 
+        waterbodies_df = pd.merge(
```

Comment:
This checks out to me as it is just the same thing you have in main at line 508.